awspub/common.py

```diff
@@ -69,3 +70,26 @@ def _get_regions(region_to_query: str, regions_allowlist: List[str]) -> List[str
         regions = ec2_regions_all
 
     return regions
+
+
+def _get_client(service, **kwargs):
+    """
+    Get a boto3 client for the given service with provided arguments
+
+    if AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables are set,
+    those will be used to create the client
+    """
+
+    aws_access_key_id = os.getenv("AWS_ACCESS_KEY_ID")
+    aws_secret_access_key = os.getenv("AWS_SECRET_ACCESS_KEY")
+    aws_region = os.getenv("AWS_REGION")
+    if aws_access_key_id and aws_secret_access_key:
+        kwargs["aws_access_key_id"] = aws_access_key_id
+        kwargs["aws_secret_access_key"] = aws_secret_access_key
+    # interacting with partitions outside of commercial need
+    # to explicitly set the region_name
```

Comment:
I think region name is always required, we just seldom need to specify because we use profiles that have a default region set already. As we switch to environment variables we'll need to start setting this for commercial partition as well